_posts/2022-01-10-reassign-partitions.md

```diff
@@ -0,0 +1,539 @@
+You can refer to the [Stimzi Quickstart Guide](https://strimzi.io/docs/operators/latest/quickstart.html) for installing Strimzi.
+
+You can install the Cluster Operator with any installation method you prefer.
+The Kafka cluster is then deployed with the plain listener enabled on port 9092.
+
+Example Kafka configuration with plain listener.
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: Kafka
+metadata:
+   name: my-cluster
+spec:
+   kafka:
+      version: 3.2.1
+      replicas: 3
+      listeners:
+         - name: plain
+           port: 9092
+           type: internal
+           tls: false
+         - name: tls
+           port: 9093
+           type: internal
+           tls: true
+      config:
+         offsets.topic.replication.factor: 3
+         transaction.state.log.replication.factor: 3
+         transaction.state.log.min.isr: 2
+         default.replication.factor: 3
+         min.insync.replicas: 2
+         inter.broker.protocol.version: "3.2"
+      storage:
+         type: jbod
+         volumes:
+            - id: 0
+              type: persistent-claim
+              size: 100Gi
+              deleteClaim: false
+            - id: 1
+              type: persistent-claim
+              size: 100Gi
+              deleteClaim: false
+   zookeeper:
+      replicas: 3
+      storage:
+         type: persistent-claim
+         size: 100Gi
+         deleteClaim: false
+   entityOperator:
+      topicOperator: {}
+      userOperator: {}
+```
+
+Once the cluster is running, let's deploy some topics where we will send and receive the messages.
+Here is an example topic configuration. You can change it to suit your requirements.
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaTopic
+metadata:
+  name: my-topic-two
+  labels:
+    strimzi.io/cluster: my-cluster
+spec:
+  partitions: 10
+  replicas: 3
+  config:
+    retention.ms: 7200000
+    segment.bytes: 1073741824
+    # ...
+```
+
+Finally, let us create a separate interactive pod for running the commands in.
+
+> You may ask why we are using a separate pod to run the commands? Can't we just use one of the broker pods?
+> The issue is that running commands from within a broker is not good practice.
+> Running any of the Kafka `/bin` scripts from within the broker container will start another JVM (with all the same settings as the Kafka broker).
+> This can cause disruption, including causing the container to run out of memory.
+> So it is always better to avoid running the commands from a broker pod.
+
+You can use the following command to create a pod called `my-pod`:
+
+```sh
+kubectl run --restart=Never --image=quay.io/strimzi/kafka:0.29.0-kafka-3.2.0 my-pod -- /bin/sh -c "sleep 3600"
+```
+
+Wait till the pod gets into the `Ready` state.
+Once it is ready, we're all set to work through the examples.
+
+## Example 1: Decreasing replication factor
+
+When we just created the `my-topic-two` topic we specified `spec.replicas` as 4.
```

Comment:
We actually did not. It has 3 replicas in the YAML above. and even if it did had 4, with 3 brokers it would probably nto create at all. Do you want to create the cluster with 5 replicas and the topic with 4 replicas to make it match?

Reply:
Yes. Sorry, it was my fault I didn't changed the replicas in the Kafka CR